### ISSUE TITLE
Inspection is not detected for `&&`(`||`) expression in `let...else`

### ISSUE THREAD

**Issue by mili-l** [intellij-rust/intellij-rust#10257]
## Environment

* **IntelliJ Rust plugin version:** 0.4.191.5268-231-nightly
* **Rust toolchain version:** 1.69.0-nightly (bd39bbb4b 2023-02-07) x86_64-apple-darwin
* **IDE name and version:** CLion 2023.1 EAP (CL-231.8109.4)
* **Operating system:** macOS 12.6
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr, org.rust.cargo.emulate.terminal, org.rust.macros.proc

## Problem description
a `&&`(`||`) expression cannot be directly assigned in `let...else`

## Steps to reproduce
```rust
    let true = true && false else { return };
    let true = true || false else { return };
```

## Actual Results
Inspection is not detected for [`&&`(`||`) expression](https://rust-lang.github.io/rfcs/3137-let-else.html#rationale-and-alternatives)
